src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -87,7 +87,9 @@ impl ManagedBarrierState {
     fn may_notify(&mut self, curr_epoch: u64) {
         // Report if there's progress on the earliest in-flight barrier.
         if self.epoch_barrier_state_map.keys().next() == Some(&curr_epoch) {
-            GLOBAL_STREAMING_METRICS.barrier_manager_progress.inc();
+            if let Some(metrics) = GLOBAL_STREAMING_METRICS.get() {
+                metrics.barrier_manager_progress.inc();
+            }
```

Comment:
cc. @zwang28 This is a side effect of changing `GLOBAL_STREAMING_METRICS` from `LazyLock` to `OnceLock`. It's difficult to tell whether `GLOBAL_STREAMING_METRICS` is guaranteed to be initialized, so we might have to put a `if let` here to check its availability, which seems a little bit ugly.